### ISSUE TITLE
AWS Console assets from CDN are blocked because of low lms score

### ISSUE THREAD

**Issue by northys** [safing/portmaster#121]
**Checklist**:

- [x] I'm using the official portmaster release (i.e no custom builds)
- [x] I searched for similar/existing issues first.
- [x] My issue is not mentioned in the Known Issues section of my [OS](https://github.com/safing/portmaster/wiki)

**What happened**:

Tried to access aws console from both firefox (deb) and chromium (snap). When I logged successfully the js assets did not load.

**What you expected to happen**:

aws console assets not being blocked

**How to reproduce it (as minimally and precisely as possible)**:

```
northys at northys-laptop in ~
» curl https://d2q66yyjeovezo.cloudfront.net/js/mezz-35gbnwov975sjdhcbsf/mezz-302ea86b8673e28960dc31df5e965a17d2c64474.gz.js
curl: (7) Failed to connect to d2q66yyjeovezo.cloudfront.net port 443: Connection refused
northys at northys-laptop in ~
» dig d2q66yyjeovezo.cloudfront.net

; <<>> DiG 9.16.1-Ubuntu <<>> d2q66yyjeovezo.cloudfront.net
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 3616
;; flags: qr rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;d2q66yyjeovezo.cloudfront.net.	IN	A

;; ANSWER SECTION:
d2q66yyjeovezo.cloudfront.net. 0 IN	A	0.0.0.0

;; Query time: 256 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Čt srp 06 18:57:35 CEST 2020
;; MSG SIZE  rcvd: 92
```

**Anything else we need to know?**:

**Environment**:

```
LOGLEVEL=debug
PORTMASTER_ARGS="--experimental-nfqueue"
```

Portmaster Version: 

```
Name: Portmaster
Version: 0.4.18
Commit: tags/v0.4.18-0-gdefd88d3a0bf4932b9152f8549d5ee39d44c4c51-dirty
Build Options: main.go
Build User: user
Build Host: docker
Build Date: 31.07.2020
Build Source: https://github.com/safing/portmaster
```

Operating System:
- [ ] Windows 7
- [ ] Windows 8/8.1
- [ ] Windows 10
- [x] Linux

```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

If applicable you can provide related sections from the log files and ensure to **remove sensitive or otherwise private information**.

```
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.310 connection:235 ▶ INFO 755 filter: dropping connection Unknown::-1 <- 192.168.31.150, endpoint is not whitelisted (incoming is always default=block)
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.319 connection:207 ▶ INFO 756 filter: granting connection root:/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-4-18:1557 -> 192.168.31.1, internally approved
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.457 terception:110 ▶ DEBU 757 accepting local dns: OUT UDP 127.0.0.1:42116 <-> 127.0.0.53:53
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.507 exp/packet:076 ▶ WARN 758 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.557 exp/packet:076 ▶ WARN 759 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.608 exp/packet:076 ▶ WARN 760 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.658 exp/packet:076 ▶ WARN 761 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.708 exp/packet:076 ▶ WARN 762 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.759 exp/packet:076 ▶ WARN 763 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.809 exp/packet:076 ▶ WARN 764 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.859 exp/packet:076 ▶ WARN 765 nfqexp: failed to set verdict AcceptAlways for 2545 (127.0.0.1 -> 127.0.0.53): netlink send: sendmsg: i/o timeout
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.868 nameserver:220 ▶ WARN 766 nameserver: possible data tunnel by northys:/snap/chromium/1244/usr/lib/chromium-browser/chrome:6845: d2q66yyjeovezo.cloudfront.net. has lms score of 7.142857, returning nxdomain
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.868 connection:221 ▶ INFO 767 filter: blocking connection northys:/snap/chromium/1244/usr/lib/chromium-browser/chrome:6845 to d2q66yyjeovezo.cloudfront.net. (<nil>), Possible data tunnel
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.961 ocess/find:023 ▶ DEBU 768 process: failed to find PID of connection: could not find connection in system state tables
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.961 connection:106 ▶ DEBU 769 network: failed to find process of packet OUT TCP ::1:33038 <-> ::1:9229: could not find connection in system state tables
srp 06 18:50:23 northys-laptop portmaster-start[1374]: 200806 18:50:23.961 connection:207 ▶ INFO 770 filter: granting connection Unknown::-1 -> ::1, endpoint is not blacklisted (default=permit)
srp 06 18:50:24 northys-laptop portmaster-start[1374]: 200806 18:50:24.072 all/master:108 ▶ WARN 771 filter: failed to find local peer process PID: could not find connection in system state tables
srp 06 18:50:24 northys-laptop portmaster-start[1374]: 200806 18:50:24.072 connection:207 ▶ INFO 772 filter: granting connection northys:/snap/chromium/1244/usr/lib/chromium-browser/chrome:6845 -> 127.0.0.1, endpoint is not blacklisted (default=permit)
srp 06 18:50:24 northys-laptop portmaster-start[1374]: 200806 18:50:24.080 all/master:108 ▶ WARN 773 filter: failed to find local peer process PID: could not find connection in system state tables
srp 06 18:50:24 northys-laptop portmaster-start[1374]: 200806 18:50:24.080 connection:207 ▶ INFO 774 filter: granting connection northys:/snap/chromium/1244/usr/lib/chromium-browser/chrome:6845 -> 127.0.0.1, endpoint is not blacklisted (default=permit)
srp 06 18:50:24 northys-laptop portmaster-start[1374]: 200806 18:50:24.080 all/master:108 ▶ WARN 775 filter: failed to find local peer process PID: could not find connection in system state tables
```

**Comment by ppacher** [comment]
Hi @northys,

thanks for reporting that! We should probably either increase the LMS threshold or let the user configure it. That "tunnel" detection is very alpha anyway. We'll discuss a solution for that and I'll report back to you.

Until then, you should be able to workaround that by explicitly whitelisting the affected domains.
Add the following to the **Endpoint Filter List** setting (it's under Privacy Filter):

```
+ *.cloudfront.net.
```

See the *Input Help* next to the setting for more information.

**Comment by northys** [comment]
> Until then, you should be able to workaround that by explicitly whitelisting the affected domains.
> Add the following to the **Endpoint Filter List** setting (it's under Privacy Filter):
> 
> ```
> + *.cloudfront.net.
> ```

I've tried to set multiple syntax to get cloudfront.net to allow list but it doesn't work for me. Still the same error in log.

```
 Format:
	Permission:
		"+": permit
		"-": block
	Host Matching:
		IP, CIDR, Country Code, ASN, Filterlist, Network Scope, "*" for any
		Domains:
			"example.com": exact match
			".example.com": exact match + subdomains
```

Man page says it should be `+ .cloudfront.net` but it also doesn't work.

**Comment by ppacher** [comment]
Hi @northys,

Thanks for reporting. Unfortunately the explicit whitelisting I mentioned didn't work for the LMS scoring. I just pushed #124 to fix that so the workaround can be applied.

**Comment by dhaavi** [comment]
Fix released with v0.5.2. Please reopen, if this does not fix the issue for you. Additionally, this check can now be turned off via a feature flag, and the endpoint list override actually works.
